zuchini-model/src/main/java/org/zuchini/parser/ThrowingErrorListener.java

```diff
@@ -0,0 +1,17 @@
+package org.zuchini.parser;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+
+class ThrowingErrorListener extends BaseErrorListener {
+
+    public static final ThrowingErrorListener INSTANCE = new ThrowingErrorListener();
+
+    @Override
+    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e)
+            throws ParseCancellationException {
+        throw new ParseCancellationException("line " + line + ":" + charPositionInLine + " " + msg);
```

Comment:
Wrap the `RecognitionException`?

Reply:
Just tried this to see if there is useful information in there, but it turns out the exception parameter is actually `null` for the missing `EOL` use case. Maybe it will be set for other errors.